src/Util/ClassSourceManipulator.php

```diff
@@ -579,10 +579,10 @@ private function addSingularRelation(BaseRelation $relation): void
 
         $this->addGetter(
             $relation->getPropertyName(),
-            $relation->getCustomReturnType() ?: $typeHint,
+            $relation->getCustomReturnType() ?? $typeHint,
```

Comment:
I believe it still is because the return type for `BaseRelation::getCustomReturnType(): ?string`. 

Before, we just had a "string" type hint instead of an actual return type. Now, we have to watch out for uninitialized properties. Side note: we're doing a lot of cleanup in CSM in the respective PR.